### PR TITLE
exports for manual mappings and improve geometry hit rate

### PR DESCRIPTION
**Pull request by damonmcc**
## motivations
- before seeking help with manual geometry creation, we'd like to maximize the number of records with geometries assigned by the build process
- this process involves geocoding, matching points to DoiTT footprint geometriesr, and matching place names to facilities/parks place names with geometries
- at the moment, the geometry hit rate is **55% geometry hit rate** (records with a geometry / records with a valid location)
- consider [Amanda's thoughts on potential improvements](https://github.com/nycplanning/db-cbbr/issues/27#issuecomment-1442197109)

## changes
- fix errors and improve clarity of geocoding
  - Geosupport function 3 was accidentally using function 2
  - generate geometries based on Geosupport function 3 results
  - replace all uses of `np.nan` and empty strings with `None` for simplicity and sql compatibility (e.g. python's `None` becomes sql's `NULL`)
- improve csv exports by removing all line breaks in input data and confirming quotes surround text that has a comma in it
- for all records without geometries, split them by priority and export as separate files to guide manual mapping efforts:
  - Priority A: `where type_br = 'C' and type = 'site'`
  - Priority B: `where type_br = 'C'`
  - Priority C: all others
- move and format unused python files to a `DEPRECATED` folder
- improve python file structures to improve relative imports (e.g. `from ..python.geocode_utils import parse_location` becomes from library.helper.geocode_utils import parse_location)
- improve python tests and run them via github action on all branches

## notes
- lots of apparent changes here. if it's helpful to highlight, all tests are passing here and, especially in the case of geocoding, they're thoroughly comparing outputs to [expected values](https://github.com/NYCPlanning/db-cbbr/blob/dm/improve-geometry-rate/cbbr_build/tests/test_data/geocoded_cbbr_data.csv)

### CODE REVIEW

**Comment by damonmcc** [comment]
current status of outputs:
- 3728 records in total
- 1290 records with a valid location (not null, not district wide)
- 996 records with spatial data (either from geocoding or matching to parks/facilities data)
- 966 records with a geometry
- 75% geometry hit rate (records with a geometry / records with a valid location)

**Comment by damonmcc** [comment]
records in need of geometries by priority level
- A: 229
- B: 950
- C: 1583

**Comment by damonmcc** [comment]
current status of outputs:
- 3728 records in total
- 1307 records with a specified location (not null, not district wide)
- 939 records with a geometry
- 72% geometry hit rate (records with a geometry / records with a specified location)

records in need of geometries by priority level
- Priority A: 236
- Priority B: 962
- Priority C: 1592
- Total: 2790

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
+name: Test
```
Comment (mbh329) [inline .github/workflows/test.yml:1]: Thinking about naming here - do we think it valuable to have a more specific title than `test`? Maybe `lint_geosupport_unit_tests`?

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Setup python
+        uses: actions/setup-python@v4.5.0
+        with:
+          python-version: "3.11.1"
+
+      - name: Setup linting
+        run: pip install black==22.8.0
```
Comment (mbh329) [inline .github/workflows/test.yml:21]: Enjoying the new test workflow, can be a model going forward for our other data products. Can add this/a version of this to the template repo

File: cbbr_build/library/geocode.py [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+    geo_function_result["geo_from_y_coord"] = geo_from_coords.get("Y Coordinate", "")
+
+    geo_from_coords = geo_client["2"](node=geo_to_node).get("SPATIAL COORDINATES", {})
+    geo_function_result["geo_to_x_coord"] = geo_from_coords.get("X Coordinate", "")
+    geo_function_result["geo_to_y_coord"] = geo_from_coords.get("Y Coordinate", "")
+
+    return geo_function_result
+
+
+GEOSUPPORT_FUNCTION_HIERARCHY = [
```
Comment (mbh329) [inline cbbr_build/library/geocode.py:108]: could throw this list in the geocode_utils.py

File: README.md [hunk 4]
```diff
@@ -39,4 +41,28 @@ CBBR is primarily built for planning coordination and information purposes only
 2. `./02_cbbr.sh` to normalize the agency and community district fields
 3. `./03_spatial.sh` to geocode the dataset
 4. `./04_export.sh` to export the dataset as csv
-5. `./05_archive.sh` to archive the cbbr to EDM postgresDB
\ No newline at end of file
+5. `./05_archive.sh` to archive the cbbr to EDM postgresDB
+
```
Comment (mbh329) [inline README.md:42]: Do we want to be specifc about setting up the pyenv environments here?

File: cbbr_build/library/geocode.py [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+    """2 function - geocode intersection based on the two street names (primary and 1st cross street)"""
+    borough = input_record.get("borough_code")
+    street_name = input_record.get("street_name")
+    cross_street_1 = input_record.get("between_cross_street_1")
+    cross_street_2 = input_record.get("and_cross_street_2")
+    if cross_street_2 and cross_street_1 != cross_street_2:
+        # non-null value for cross_street_2 that isn't the same as cross_street_1
+        # indicates a likely street segment
+        # NOTE This error message won't show in data since, if no function work, geo_message will be GEOCODING FAILED
+        raise GeosupportError("UNLIKELY TO BE AN INTERSECTION")
```
Comment (mbh329) [inline cbbr_build/library/geocode.py:65]: the geosupport errors are helpful!

**Review by mbh329 — APPROVED** [review]
Looks good to me - had some minor comments related to content but ran through the build process locally and got the expected outputs on my machine

File: README.md (hunk 4 above)
Comment (damonmcc) [reply README.md:42]: clarified in [233de9d](https://github.com/NYCPlanning/db-cbbr/pull/31/commits/233de9d0e7c8da324d061a833880535e2bf27e92)

File: .github/workflows/test.yml (hunk 1 above)
Comment (damonmcc) [reply .github/workflows/test.yml:1]: the individual jobs in the workflow actually do have the more specific names. so they show up like this:
image

**Pull request merged**